From 88ca64946f87fd318dc9e9d25a289787428a2642 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Tue, 27 Sep 2016 14:03:08 +0300
Subject: [PATCH 1109/2241] dts: ap806: add description for ap806-clk driver

Add device tree settings for AP806 CPU clock driver,
including CPU cluster assignment.

Change-Id: Ifcae10b9727935d816e81c8dadcebbb7db7b9df0
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33197
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi | 2 ++
 arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi | 4 ++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      | 8 ++++++++
 3 files changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
index 95a1ff6..8b405e9 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
@@ -58,12 +58,14 @@
 			device_type = "cpu";
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x000>;
+			clocks = <&ap806_clock 0>;
 			enable-method = "psci";
 		};
 		cpu@001 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x001>;
+			clocks = <&ap806_clock 0>;
 			enable-method = "psci";
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
index 3ff612e..8191fde 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
@@ -58,24 +58,28 @@
 			device_type = "cpu";
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x000>;
+			clocks = <&ap806_clock 0>;
 			enable-method = "psci";
 		};
 		cpu1: cpu@001 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x001>;
+			clocks = <&ap806_clock 0>;
 			enable-method = "psci";
 		};
 		cpu2: cpu@100 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x100>;
+			clocks = <&ap806_clock 1>;
 			enable-method = "psci";
 		};
 		cpu3: cpu@101 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x101>;
+			clocks = <&ap806_clock 1>;
 			enable-method = "psci";
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 6331536..38e4459 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -167,6 +167,14 @@
 					     <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
 			};
 
+			ap806_clock: ap806_clock@6f8200 {
+				compatible = "marvell,ap806-cpu-clk", "syscon";
+				#clock-cells = <1>;
+				clocks = <&syscon 0>;
+				reg = <0x6f8200 0x100>;
+				status = "okay";
+			};
+
 			axim-ddr-rd@840000 {
 				compatible = "marvell,coresight-axim", "arm,primecell";
 				reg = <0x840000 0x1000>;
-- 
2.7.4

